functions/src/scripts/denormalize-avatar-urls.ts

```diff
@@ -0,0 +1,133 @@
+// Script for lining up users and contracts/comments to make sure the denormalized avatar URLs in the contracts and
+// comments match the user avatar URLs.
+
+import * as admin from 'firebase-admin'
+import { initAdmin } from './script-init'
+import {
+  DocumentCorrespondence,
+  findDiffs,
+  describeDiff,
+  applyDiff,
+} from './denormalize'
+import { DocumentSnapshot, Transaction } from 'firebase-admin/firestore'
+
+initAdmin()
+const firestore = admin.firestore()
+
+async function getUsersById(
+  transaction: Transaction
+): Promise<Map<string, DocumentSnapshot>> {
```

Comment:
You typically don't need to specify return types.

In this case it probably can't infer the return type, but that's because your new Map() is missing type parameters:

`const results = new Map<string, DocumentSnapshot>()`